### PR TITLE
BL-3983 Message on Copy failed

### PR DESCRIPTION
**Pull request by gmartin7**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/1326)

### CODE REVIEW

File: src/BloomExe/Edit/EditingView.cs [hunk 1]
```diff
@@ -727,13 +728,22 @@ private static bool CopyImageToClipboard(DomEventArgs ge, string bookFolderPath)
...
+					using (var image = RobustIO.PalasoImageFromFile(path))
 					{
 						BloomClipboard.CopyImageToClipboard(image);
 					}
 					return true;
 				}
-				catch(Exception e)
+				catch (ExternalException e)
+				{
+					Logger.WriteEvent("CopyImageToClipboard -> ExternalException: " + e.Message);
```
Comment (hatton) [inline src/BloomExe/Edit/EditingView.cs]: I suspect it doesn't work to put newlines in a string that will be localized; we won't have separate Windows and Linux tmx files. Instead, you'd need to get the localized string and *then* replace newlines with the one needed for this string.

**Comment by StephenMcConnel** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Edit/EditingView.cs, line 739 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1326#-KWtRwYUoo10k5v5wrLx-r1-739:-KWtlYuzGKC4rcEFpzc6:b-cark6e) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/7195dc90e8f5f2e8c58ce2682dc0fea931f40962/src/BloomExe/Edit/EditingView.cs#L739)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary>
> I suspect it doesn't work to put newlines in a string that will be localized; we won't have separate Windows and Linux tmx files. Instead, you'd need to get the localized string and *then* replace newlines with the one needed for this string.
</details>
A common practice is to use String.Format with {n} (for some number n) being replaced by Environment.NewLine.  A final argument to LocalizationManager.GetDynamicString() is a comment for the translater that can used to describe what {0} {1} etc are filled in with.  Using Environment.NewLine in the middle of the string doesn't work as well as you might think when it comes to something that will be localized.  :-)  Although it would work to join two independently localized sentences/paragraphs.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1326)*

**Comment by hatton** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Edit/EditingView.cs, line 739 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1326#-KWtRwYUoo10k5v5wrLx-r1-739:-KWtnqgWTNOH2Q_-tjPI:bn3ky4x) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/7195dc90e8f5f2e8c58ce2682dc0fea931f40962/src/BloomExe/Edit/EditingView.cs#L739)):*
<details><summary><i>Previously, StephenMcConnel (Steve McConnel) wrote…</i></summary>
> A common practice is to use String.Format with {n} (for some number n) being replaced by Environment.NewLine.  A final argument to LocalizationManager.GetDynamicString() is a comment for the translater that can used to describe what {0} {1} etc are filled in with.  Using Environment.NewLine in the middle of the string doesn't work as well as you might think when it comes to something that will be localized.  :-)  Although it would work to join two independently localized sentences/paragraphs.
</details>
I wonder if we should leave it to the translator to insert paragraphs or not as they see fit, as a general rule.   Generally I avoid using MessageBox... our own competitors could easily be made to always use the appropriate line endings, so we don't have to think about that each time we build a string.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1326)*

**Comment by gmartin7** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Edit/EditingView.cs, line 739 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1326#-KWtRwYUoo10k5v5wrLx-r1-739:-KX7Ev-PoY3XUFthZ3EY:b-pkbqyq) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/7195dc90e8f5f2e8c58ce2682dc0fea931f40962/src/BloomExe/Edit/EditingView.cs#L739)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary>
> I wonder if we should leave it to the translator to insert paragraphs or not as they see fit, as a general rule.   Generally I avoid using MessageBox... our own competitors could easily be made to always use the appropriate line endings, so we don't have to think about that each time we build a string.
</details>
I split the message into two parts. It's conceivable that we might use the 2nd part by itself someday anyway.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1326)*

**Comment by andrew-polk** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1326#-:-KX7K8x_oiJ46Epy75UP:bnfp4nl)*

**Pull request merged**